pkg/stanza/fileconsumer/finder_test.go

```diff
@@ -123,6 +124,206 @@ func TestFinder(t *testing.T) {
+			expected: []string{"err.2023020612.log"},
+		},
+		{
+			name:    "Timestamp Sorting Ascending",
+			files:   []string{"err.2023020612.log", "err.2023020611.log", "err.2023020610.log", "err.2023020609.log"},
+			include: []string{"err.*.log"},
+			exclude: []string{},
+			filterSortRule: OrderingCriteria{
+				Regex: `err\.(?P<value>\d{4}\d{2}\d{2}\d{2}).*log`,
+				SortBy: []SortRuleImpl{
+					{
+						TimestampSortRule{
+							BaseSortRule: BaseSortRule{
+								RegexKey:  "value",
+								Ascending: true,
+							},
+							Location: "UTC",
+							Format:   `%Y%m%d%H`,
+						},
+					},
+				},
+			},
+			expected: []string{"err.2023020609.log"},
+		},
+		{
+			name:    "Numeric Sorting",
+			files:   []string{"err.123456789.log", "err.123456788.log", "err.123456787.log", "err.123456786.log"},
+			include: []string{"err.*.log"},
+			exclude: []string{},
+			filterSortRule: OrderingCriteria{
+				Regex: `err\.(?P<value>\d+).*log`,
+				SortBy: []SortRuleImpl{
+					{
+						NumericSortRule{
+							BaseSortRule: BaseSortRule{
+								RegexKey:  "value",
+								Ascending: false,
+							},
+						},
+					},
+				},
+			},
+			expected: []string{"err.123456789.log"},
+		},
+		{
+			name:    "Numeric Sorting Ascending",
+			files:   []string{"err.123456789.log", "err.123456788.log", "err.123456787.log", "err.123456786.log"},
+			include: []string{"err.*.log"},
+			exclude: []string{},
+			filterSortRule: OrderingCriteria{
+				Regex: `err\.(?P<value>\d+).*log`,
+				SortBy: []SortRuleImpl{
+					{
+						NumericSortRule{
+							BaseSortRule: BaseSortRule{
+								RegexKey:  "value",
+								Ascending: true,
+							},
+						},
+					},
+				},
+			},
+			expected: []string{"err.123456786.log"},
+		},
+		{
+			name:    "Alphabetical Sorting",
+			files:   []string{"err.a.log", "err.b.log", "err.c.log", "err.d.log"},
+			include: []string{"err.*.log"},
+			exclude: []string{},
+			filterSortRule: OrderingCriteria{
+				Regex: `err\.(?P<value>[a-zA-Z]+).*log`,
+				SortBy: []SortRuleImpl{
+					{
+						AlphabeticalSortRule{
+							BaseSortRule: BaseSortRule{
+								RegexKey:  "value",
+								Ascending: false,
+							},
+						},
+					},
+				},
+			},
+			expected: []string{"err.d.log"},
+		},
+		{
+			name:    "Alphabetical Sorting Ascending",
+			files:   []string{"err.a.log", "err.b.log", "err.c.log", "err.d.log"},
+			include: []string{"err.*.log"},
+			exclude: []string{},
+			filterSortRule: OrderingCriteria{
+				Regex: `err\.(?P<value>[a-zA-Z]+).*log`,
+				SortBy: []SortRuleImpl{
+					{
+						AlphabeticalSortRule{
+							BaseSortRule: BaseSortRule{
+								RegexKey:  "value",
+								Ascending: true,
+							},
+						},
+					},
+				},
+			},
+			expected: []string{"err.a.log"},
+		},
+		{
+			name:    "Multiple Sorting - timestamp priority sort",
+			files:   []string{"err.a.123456786.2023020610.log", "err.b.123456789.2023020609.log", "err.c.123456788.2023020612.log", "err.d.123456787.2023020611.log"},
```

Comment:
For this one:

```
"err.b.1.2023020601.log"
"err.b.2.2023020601.log"
"err.a.1.2023020601.log"
"err.a.2.2023020601.log"
"err.b.1.2023020602.log"
"err.a.2.2023020602.log"
"err.b.2.2023020602.log" // expected
"err.a.1.2023020602.log"
```

`a|b` descending:

```
"err.b.1.2023020601.log"
"err.b.2.2023020601.log"
"err.b.1.2023020602.log"
"err.b.2.2023020602.log" // expected
"err.a.1.2023020601.log"
"err.a.2.2023020601.log"
"err.a.2.2023020602.log"
"err.a.1.2023020602.log"
```

then `1|2` descending:

```
"err.b.2.2023020601.log"
"err.b.2.2023020602.log" // expected
"err.a.2.2023020601.log"
"err.a.2.2023020602.log"
"err.b.1.2023020601.log"
"err.b.1.2023020602.log"
"err.a.1.2023020601.log"
"err.a.1.2023020602.log"
```

finally `2023..01|2023..02` descending:

```
"err.b.2.2023020602.log" // expected
"err.a.2.2023020602.log"
"err.b.1.2023020602.log"
"err.a.1.2023020602.log"
"err.b.2.2023020601.log"
"err.a.2.2023020601.log"
"err.b.1.2023020601.log"
"err.a.1.2023020601.log"
```

Reply:
I've attempted to do the same with this one.